common/serde-reflection/src/trace.rs

```diff
@@ -54,11 +52,26 @@ impl SerializationRecords {
     }
 }
 
+/// Configuration object to create a tracer.
+#[derive(Debug)]
+pub struct TracerConfig {
+    pub is_human_readable: bool,
```

Comment:
Might want to be careful about types with all public fields in the public API since any additions become breaking changes. Might want to stick a `#[nonexhaustive]` on top of this.

Reply:
TIL! This is really nice.

Reply:
Except it doesn't work with functional record updates :(
https://internals.rust-lang.org/t/proto-rfc-expanded-functional-record-update/3077

Reply:
I made the fields private and added "builder-style" setters in the end.